awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
>  Do you think we should start using it for both standard in and out? 

I don't think we need to migrate everything over to `BinaryStdout` right now, but as we're consolidating things over to `compat.py`, I don't see any reason to have different ways of doing binary IO depending on if you're using output vs. input.  We should at least have a consistent way of doing things.

I see this as an opportunity to clean up technical debt.